src/store/preferences.ts

```diff
@@ -103,16 +105,21 @@ export const actions = {
 
     await dispatch(
       'preferences/fetchPreferences',
-      {
-        port, user, password
-      },
+      args,
       { root: true });
   },
 
-  updatePreferencesData<P extends RecursiveKeys<Settings>>({ commit, state }: PrefActionContext, args: {property: P, value: RecursiveTypes<Settings>[P]}) {
+  updatePreferencesData<P extends RecursiveKeys<Settings>>({
```

Comment:
Could you add a jsdoc explaining what's going on here? There are a few abstraction layers going on with the template. Specifically, what is the call to `dispatch` supposed to do?  Also, should we specify the return-type of this function?

Reply:
Things like `commit` and `dispatch` are vuex implementations

https://v3.vuex.vuejs.org/guide/mutations.html
https://v3.vuex.vuejs.org/guide/actions.html

I added some docs to the action here. Let me know what you think